### PR TITLE
Refactor Arduino Code

### LINKED ISSUE #19: [ESP32] De-dupe and refactor the code
De-dupe and refactor the code, I copy-pasted quite a bit in loop() to handle different states.

I'm sure there is a ton of cleanup that could be done to make the code more understandable and easier to maintain.

### PR DESCRIPTION
**Pull request by wsciaroni**
Closes #19

This MR seeks to simply move the code into functions.

I do not have hardware to test this.

I made a few functions that should reduce some duplication across the code. I plan to make a second pull request where I will try to make this a bit more object oriented.

### CODE REVIEW

File: microcontroller-src/kv4p_ht_esp32_wroom_32/kv4p_ht_esp32_wroom_32.ino [hunk 1]
```diff
@@ -161,362 +165,381 @@ void initI2SRx() {
...
+  case COMMAND_GET_FIRMWARE_VER:
+  {
+    Serial.write(VERSION_PREFIX, sizeof(VERSION_PREFIX));
+    Serial.write(FIRMWARE_VER, sizeof(FIRMWARE_VER));
+    Serial.flush();
+    esp_task_wdt_reset();
+    return;
+  }
+  break;
+  // TODO get rid of the code duplication here and in MODE_RX below to handle COMMAND_TUNE_TO and COMMAND_FILTERS.
```
Comment (VanceVagell) [inline microcontroller-src/kv4p_ht_esp32_wroom_32/kv4p_ht_esp32_wroom_32.ino:260]: I think you addressed this TODO and can remove it now.

**Comment by VanceVagell** [comment]
I see this still says WIP but it looks in decent shape. Are you planning to make more changes in this?

**Comment by wsciaroni** [comment]
> I see this still says WIP but it looks in decent shape. Are you planning to make more changes in this?

I have branched from here to start a more thorough overhaul. The follow-on is #40


Other than that, this is completely untested, so please test it before merging.

**Comment by VanceVagell** [comment]
Per discussion with wsciaroni, this is code complete but needs to be tested on device (he does not yet have the board to test it himself).

**Comment by wsciaroni** [comment]
Instead of doing a partial upgrade with this PR, we will instead spend time on #40

**Pull request closed without merging**